src/run_tribler.py

```diff
@@ -88,10 +89,21 @@ async def start_tribler():
     get_event_loop().run_forever()
 
 
+def excepthook(exctype, value, traceback):  # pylint: disable=unused-argument
+    ignored_exceptions = [KeyboardInterrupt, ]
+    ignored_exceptions.extend(IGNORED_ERRORS)
+    if exctype in ignored_exceptions:
+        return
```

Comment:
`IGNORED_ERRORS` contains both exception classes and tuples of exception classes and error codes:

https://github.com/Tribler/tribler/blob/63dc402cb4def5743099e1789fefd07942f9d5ea/src/tribler-core/tribler_core/session.py#L63-L76

Your code would ignore all tuple entries (essentially only checking for `gaierror`). If this is intended please add a line comment. If it is unintended.. there is probably some way to get the error code from the `traceback` 🤷.

I would also suggest adding [SystemExit](https://docs.python.org/3/library/exceptions.html#SystemExit) to the ignored exceptions - it is the second "special" type of Python exception.

Reply:
🤦 thanks

Reply:
At first, I want to add all errors (without codes) from `IGNORED_ERRORS', but it is not necessary. 
`session.unhandled_error_observer(self, loop, context)` should eat them anyway.

I've added `SystemExit` and remove `ignored_exceptions.extend(IGNORED_ERRORS)`.

Thank you for pointing this out.